splice_spark2/src/main/java/com/splicemachine/nsds/kafka/KafkaMaintenance.java

```diff
@@ -96,19 +100,43 @@ public static void deleteOldTopics(String kafkaServers, String dataFileName, lon
 
     /**
      * 
-     * @param args args[0] Kafka Bootstrap Servers; args[1] path to topics file
+     * @param args args[0] Kafka Bootstrap Servers; args[1] path to topics file.
+     *             Optional: args[3] Topic age cutoff in minutes
+     *             args[4] Comma-separated list of topic prefixes to delete, topics with other prefixes will not be deleted
+     *             args[5] Comma-separated list of topic prefixes to keep, topics with other prefixes will be deleted
      */
     public static void main(String[] args) {
         try {
             if( args.length < 2 ) {
-                System.out.println("Required params: Kafka Bootstrap Servers, Path to topics file. Optional: Topic age cutoff in minutes.");
+                System.out.println(
+                    "Required params: Kafka Bootstrap Servers, Path to topics file."
+                    + "  Optional: Topic age cutoff in minutes,"
+                    + " Comma-separated list of topic prefixes to delete,"
+                    + " Comma-separated list of topic prefixes to keep."
+                );
                 System.exit(1);
             }
             long ageCutoffMinutes = 24*60;
             if( args.length >= 3 ) {
                 ageCutoffMinutes = Long.parseLong( args[2] );
             }
-            KafkaMaintenance.deleteOldTopics( args[0] , args[1] , ageCutoffMinutes );
+            List<String> prefixToDelete = null;
+            if( args.length >= 4 && !args[3].isEmpty() ) {
+                prefixToDelete = Arrays.asList( args[3].split(",") );
+            }
+            List<String> prefixToKeep = null;
+            if( args.length >= 5 && !args[4].isEmpty() ) {
+                if( !args[3].isEmpty() && !args[4].isEmpty() ) {
```

Comment:
Remove redundant condition in inner if block: !args[4].isEmpty()

Reply:
Fixed in commit https://github.com/splicemachine/spliceengine/pull/4991/commits/d04691e8f2c188cfdb70f38c43a4b57835318f8f